autoload/gitmessenger/git.vim

```diff
@@ -8,7 +8,7 @@ let s:SEP = has('win32') ? '\' : '/'
 "     empty string means root directory was not found
 function! gitmessenger#git#root_dir(from) abort
     let from = fnamemodify(a:from, ':p')
-    if from[-1] ==# s:SEP
+    if from[-1:] ==# s:SEP
```

Comment:
This `if` condition checks the last character of `from` is `\`. So `from[-1] ==# s:SEP` is correct. And I confirmed this plugin worked on Windows. Please file a ticket to understand what is going wrong on your environment.